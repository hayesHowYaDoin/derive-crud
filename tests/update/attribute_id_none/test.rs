use derive_crud::Update;

#[derive(Update)]
#[crud_table("test_table")]
struct Test {
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    let _result = Test {
        id: 1,
        name: "Debbie".to_string(),
    }
    .update()
    .await;
}
