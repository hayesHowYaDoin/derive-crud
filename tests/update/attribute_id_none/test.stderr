error: A struct field must be annotated with `#[crud_id]` to specify the database table
 --> tests/update/attribute_id_none/test.rs:5:13
  |
5 |   struct Test {
  |  _____________^
6 | |     id: i64,
7 | |     name: String,
8 | | }
  | |_^

error[E0599]: no method named `update` found for struct `Test` in the current scope
  --> tests/update/attribute_id_none/test.rs:16:6
   |
5  |   struct Test {
   |   ----------- method `update` not found for this struct
...
12 |       let _result = Test {
   |  ___________________-
13 | |         id: 1,
14 | |         name: "Debbie".to_string(),
15 | |     }
16 | |     .update()
   | |     -^^^^^^ method not found in `Test`
   | |_____|
   |
